src/server/project.ts

```diff
@@ -1180,12 +1190,25 @@ namespace ts.server {
                 }
                 this.projectService.logger.info(`Plugin validation succeded`);
                 this.languageService = newLS;
-                this.plugins.push(pluginModule);
+                this.plugins.push({ name: configEntry.name, module: pluginModule });
             }
             catch (e) {
                 this.projectService.logger.info(`Plugin activation failed: ${e}`);
             }
         }
+
+        onPluginConfigurationChanged(pluginName: string, configuration: any) {

```

Comment:
Are you sure this needs to be exposed and not internal?

Reply:
Thanks I'll make it internal. Wasn't sure about the logic behind what's internal and what's public.